src/main/java/hsleiden/stenentijdperk/stenentijdperk/Controllers/BoardController.java

```diff
@@ -237,52 +240,59 @@ private void moreVillagerHut(int index) {
     }
 
     private void buttonCheckPhase1(int index) {
-        if (locatieVrij(index) && !boardmodel.getPlaced()) {
-            if (index == 6 && playercontroller.getVillagers(boardmodel.getPlayer()) >= 2) {
-                plaatsenStamleden(index, 2);
-            } else if (index != 6) {
-                plaatsenStamleden(index, 1);
+        if(FirebaseController.getBoard(String.valueOf(localPlayer.getLobby())).getPlayer().getPlayerNumber().equals(localPlayer.getPlayerNumber())) {
+            if (locatieVrij(index) && !boardmodel.getPlaced()) {
+                if (index == 6 && playercontroller.getVillagers(boardmodel.getPlayer()) >= 2) {
+                    plaatsenStamleden(index, 2);
+                } else if (index != 6) {
+                    plaatsenStamleden(index, 1);
+                }
```

Comment:
goede code, maar weer een call. Je kan mogelijk denken aan een manier om maar 1 keer per beurt een request te sturen in plaats van per klik.